libqtile/extension/base.py

```diff
@@ -18,66 +18,37 @@
 # LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 # SOFTWARE.
-import re
+
+from __future__ import annotations
+
 import shlex
 from subprocess import PIPE, Popen
-from typing import Any
+from typing import TYPE_CHECKING
 
 from libqtile import configurable
 from libqtile.log_utils import logger
 
-RGB = re.compile(r"^#?([a-fA-F0-9]{3}|[a-fA-F0-9]{6})$")
+if TYPE_CHECKING:
+    from typing import Any
```

Comment:
This seems like a step backwards; can we get rid of this and just use Optional directly if we drop support for e.g. 3.10?

Reply:
That is how it is done. In this case the TYPE_CHECKING block seems redundant, because it is only used for typing module stuff. But you have those blocks anyway in the whole project.

I feel deprecating the python version is a topic that should be discussed on a broader level.

Reply:
> But you have those blocks anyway in the whole project.

Yes, and I have been deleting them as we can; I really don't want to add more.

Reply:
In that case we can replace the typing.Optional with `x | None`, but where does `typing.Any` come from?
You could argue that we can get rid of the block in this case by default importing `typing.Any`, but getting rid
of all the `if TYPE_CHECKING` blocks is going to be very hard, iff you want to be able to strictly type check the code - my experience.

Reply:
`Any` was typically used to typing the `defaults` section.